### PR TITLE
Chapter 14 - Following Users

### PR DESCRIPTION
**Pull request by EvonneNgXinYi**
Follow and unfollow other users

### CODE REVIEW

**Comment by EvonneNgXinYi** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -155,6 +155,48 @@ aside {
...
+  overflow: auto;
+  margin-top: 0;
+  padding: 0;
+
+  a {
+    float: left;
+    padding: 0 10px;
+    border-left: 1px solid $light-gray;
+    color: $gray;
+    &:first-child {
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss]:
```suggestion
    color: $gray;

    &:first-child {
```

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -155,6 +155,48 @@ aside {
...
+
+  strong {
+    display: block;
+  }
+}
+
+.user_avatars {
+  overflow: auto;
+  margin-top: 10px;
+  .gravatar {
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss]:
```suggestion
  margin-top: 10px;

  .gravatar {
```

File: app/assets/stylesheets/relationships.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Relationships controller here.
```
Comment (chungpv-1008) [inline app/assets/stylesheets/relationships.scss]: remove this file

File: app/controllers/relationships_controller.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+class RelationshipsController < ApplicationController
+  before_action :logged_in_user
+
+  def index
+    @user = User.find_by id: params[:id]
+    redirect_to root_path unless @user
```
Comment (chungpv-1008) [inline app/controllers/relationships_controller.rb]: You should create method like this:
```
private

def load_user
  @user = User.find_by id: params[:id] || params[:followed_id]
  return if @user.present?

  flash...
  redirect_to root_path
end
```
Then create before_action for 2 action 'index' and 'create'

File: app/views/relationships/create.js.erb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+$("#follow_form").html('<%= j render "users/unfollow" %>');
+$("#followers").html('<%= @user.followers.count %>');
```
Comment (chungpv-1008) [inline app/views/relationships/create.js.erb]: using ' instead of "

File: app/views/relationships/destroy.js.erb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+$("#follow_form").html('<%= j render "users/follow" %>');
```
Comment (chungpv-1008) [inline app/views/relationships/destroy.js.erb]: similar

File: app/views/users/_follow.html.erb [hunk 7]
```diff
@@ -0,0 +1,4 @@
+<%= form_(current_user.active_relationships.build, remote: true) do |f| %>
```
Comment (chungpv-1008) [inline app/views/users/_follow.html.erb]:
```suggestion
<%= form_for current_user.active_relationships.build, remote: true do |f| %>
```

File: app/views/users/_unfollow.html.erb [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<%= form_for(current_user.active_relationships.find_by followed_id: @user.id), html: { method: :delete }, remote: true) do |f| %>
```
Comment (chungpv-1008) [inline app/views/users/_unfollow.html.erb]:
```suggestion
<%= form_for current_user.active_relationships.find_by(followed_id: @user.id), html: { method: :delete }, remote: true do |f| %>
```

**Comment by EvonneNgXinYi** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**